.github/workflows/ci-cd.yml

```diff
@@ -0,0 +1,72 @@
+name: CI/CD Pipeline
+
+on:
+  push:
+    branches: [ master, dev, feature/* ]
+  pull_request:
+    branches: [ main, dev, feature/* ]
+
+jobs:
+  build-and-test:
+    runs-on: ubuntu-latest
+    
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Setup .NET
+        uses: actions/setup-dotnet@v4
+        with:
+          dotnet-version: '8.0.x'
+
+      - name: Restore dependencies
+        run: dotnet restore Airbnb.sln
+
+      - name: Build
+        run: dotnet build Airbnb.sln --configuration Release --no-restore
+
+      - name: Run Unit Tests
+        run: dotnet test Airbnb.Application.Tests/Airbnb.Application.Tests.csproj --configuration Release --no-build --logger "trx;LogFileName=test-results.trx" --results-directory ./TestResults
+
+      - name: Publish Test Report
+        uses: actions/upload-artifact@v4
+        with:
+          name: test-results
+          path: ./TestResults
+
+      - name: Generate coverage report
+        run: |
+          dotnet tool install --global dotnet-reportgenerator-globaltool
+          dotnet test Airbnb.Application.Tests/Airbnb.Application.Tests.csproj --collect:"XPlat Code Coverage" --results-directory ./TestResults
+          reportgenerator -reports:./TestResults/**/coverage.cobertura.xml -targetdir:./TestResults/CoverageReport -reporttypes:Html
+
+      - name: Upload coverage report
+        uses: actions/upload-artifact@v4
+        with:
+          name: coverage-report
+          path: ./TestResults/CoverageReport
+  
+  docker:
+    needs: build-and-test
+    runs-on: ubuntu-latest
+    
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up Docker Builds
+        uses: docker/setup-buildx-action@v3
+
+      - name: Login to DockerHub
+        uses: docker/login-action@v3
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: Build and push Docker image
+        uses: docker/build-push-action@v5
+        with:
+          context: .
+          file: ./Dockerfile
+          push: true
+          tags: ${{ secrets.DOCKERHUB_USERNAME }}/airbnb-project:latest
```

Comment:
it'll automatically replace 'latest' tag, but previous image versions won't be available. Rework tag to be commit hash